purchase_rfq_number/models/purchase_order.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 ProThai Technology Co.,Ltd. (http://prothaitechnology.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+import base64
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    rfq_number = fields.Char(
+        "RFQ Reference",
+        index=True,
+        copy=False,
+        default="New",
+    )
+
+    @api.model
+    def create(self, vals):
+        company = False
+        if "company_id" in vals:
+            company = self.env["res.company"].browse(vals.get("company_id"))
+        else:
+            company = self.env.company
+        if not company.keep_name_po and vals.get("name", "New") == "New":
+            vals["name"] = self.env["ir.sequence"].next_by_code("purchase.rfq") or "New"
+            vals["rfq_number"] = vals["name"]
+        return super().create(vals)
+
+    def button_confirm(self):
+        for order in self:
+            if (
+                order.state not in ["draft", "sent"]
+                and not order.company_id.keep_name_po
+            ):
+                continue
+            else:
+                order.write(
+                    {
+                        "name": self.env["ir.sequence"].next_by_code("purchase.order"),
+                    }
+                )
+
+            # save rfq pdf as attachment
+            order.action_get_rfq_attachment()
```

Comment:
I think we shouldn't make attachment. Because when you say form "report_purchase_quotation", it is too specific (in real project, as user want to use other form. **Or you need to make it default but configurable in the settings menu.** 

**Note:** in report action, there is standard feature to make a pdf attachment on your first print. User will use that instead?